cranelift-codegen/src/ir/immediates.rs

```diff
@@ -355,92 +356,6 @@ impl From<&[u8]> for V128Imm {
     }
 }
 
-impl FromStr for V128Imm {
-    type Err = &'static str;
-
-    // parse a 128-bit vector from a hexadecimal string, formatted as above
-    fn from_str(s: &str) -> Result<Self, &'static str> {
-        if s.len() <= 2 || &s[0..2] != "0x" {
-            Err("Expected a hexadecimal string, e.g. 0x1234")
-        } else {
-            // clean and check the string
-            let cleaned: Vec<u8> = s[2..]
-                .as_bytes()
-                .iter()
-                .filter(|&&b| b as char != '_')
-                .cloned()
-                .collect(); // remove 0x prefix and any intervening _ characters
-
-            if cleaned.len() == 0 {
-                Err("Hexadecimal string must have some digits")
-            } else if cleaned.len() % 2 != 0 {
-                Err("Hexadecimal string must have an even number of digits")
-            } else if cleaned.len() > 32 {
-                Err("Hexadecimal string has too many digits to fit in a 128-bit vector")
-            } else {
-                let mut buffer = [0; 16]; // zero-fill the buffer
-                let mut position = cleaned.len() / 2 - 1; // since Uimm128 is little-endian but the string is not, we write from back to front but must start at the highest position required by the string
-                for i in (0..cleaned.len()).step_by(2) {
-                    let pair = from_utf8(&cleaned[i..i + 2])
-                        .or_else(|_| Err("Unable to parse hexadecimal pair as UTF-8"))?;
-                    let byte = u8::from_str_radix(pair, 16)
-                        .or_else(|_| Err("Unable to parse as hexadecimal"))?;
-                    buffer[position] = byte;
-                    position = position.wrapping_sub(1); // should only wrap on the last iteration
-                }
-
-                Ok(V128Imm(buffer))
-            }
-        }
-    }
-}
-
-/// Implement a way to convert an iterator of immediates to a Uimm128:
-///  - this expects the items in reverse order (e.g. last lane first) which is the natural output of pushing items into a vector
-///  - this may not fully consume the iterator or may fail if it cannot take the expected number of items
-///  - this requires the input type (i.e. $ty) to implement ToBytes
-macro_rules! construct_uimm128_from_iterator_of {
-    ( $ty:ident, $lanes:expr ) => {
-        impl FromIterator<$ty> for V128Imm {
```

Comment:
I added this macro to make it easier and cleaner to do what needed to be done in https://github.com/CraneStation/cranelift/pull/1098/files#diff-1cf073842b27324c2cd6cd886b4ed198R868-R875. Now, with `SizedBoolean` and the new behavior on `ConstantData` all of this can go away.